frontend/src/pages/instance/Settings/index.vue

```diff
@@ -1,17 +1,35 @@
 <template>
     <div class="mb-3">
-        <SectionTopMenu hero="Settings" info="" />
+        <SectionTopMenu hero="Settings" info="">
+            <template #tools>
+                <ff-button
+                    v-if="tools.saveButton.visible"
+                    :disabled="tools.saveButton.disabled"
+                    class="ff-btn ff-btn--primary"
+                    size="small"
+                    data-el="save-settings-button"
+                    @click="onSaveButtonClick"
+                >
+                    {{ tools.saveButton.label }}
+                </ff-button>
+            </template>
+        </SectionTopMenu>
     </div>
     <div class="flex flex-col sm:flex-row">
-        <SectionSideMenu :options="sideNavigation" />
+        <SectionSideMenu :options="navigation" />
         <div class="flex-grow">
-            <router-view
-                :project="instance"
-                :instance="instance"
-                @instance-updated="$emit('instance-updated')"
-                @instance-confirm-suspend="$emit('instance-confirm-suspend')"
-                @instance-confirm-delete="$emit('instance-confirm-delete')"
-            />
+            <router-view v-slot="{ Component }">
```

Comment:
Talk me through this change please?

Reply:
It's because I needed to place the ref on to the actual child component itself not the surrounding router-view wrapper so that I can call `this.$refs.settingsPage.saveSettings() ` in the onSaveButtonClick method which triggers the saveSettings method on the nested page.

It's not the best approach because I'm coupling the pages together but it saved me the hassle of extracting each child's save method into the parent along their state and by doing it this way it also saves me the hassle of managing state in the parent component as well